src/snekmate/auth/Ownable2Step.vy

```diff
@@ -15,14 +15,26 @@
 """
 
 
-# @dev Returns the address of the current owner.
-# @notice If you declare a variable as `public`,
-# Vyper automatically generates an `external`
-# getter function for the variable.
-owner: public(address)
+# @dev We import and use the `Ownable` module.
+from . import Ownable as ownable
+uses: ownable
+
+
+# @dev We export (i.e. the runtime bytecode exposes these
+# functions externally, allowing them to be called using
+# the ABI encoding specification) the `external` getter
+# function `owner` from the `Ownable` module.
+# @notice Please note that you must always also export (if
+# required by the contract logic) `public` declared `constant`,
+# `immutable`, and state variables, for which Vyper automatically
+# generates an `external` getter function for the variable.
+exports: ownable.owner
 
 
 # @dev Returns the address of the pending owner.
+# @notice If you declare a variable as `public`,
+# Vyper automatically generates an `external`
```

Comment:
i feel like this comments are a bit out of place as they are general information about the language rather than this implementation.

Reply:
So 🐍 snekmate should also serve as educational material. Many newcomers to Vyper don't know that getters are automatically generated, and I want to help people understand the language well by reading through my contracts. I prefer overcommunicating than undercommunicating. 🐍 snekmate contracts are full of such gems, example:

https://github.com/pcaversaccio/snekmate/blob/e1b9a8c5784c518559a9bdeb60efc0252db3759f/src/snekmate/extensions/ERC4626.vy#L1001-L1012

My personal aim is that you can simply read through 🐍 snekmate contracts without reading the docs and you understand how Vyper works :)